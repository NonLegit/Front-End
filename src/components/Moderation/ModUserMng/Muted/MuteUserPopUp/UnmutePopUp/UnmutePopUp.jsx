/* eslint-disable import/no-cycle */
import * as React from 'react';
import { Divider, Box, Typography } from '@mui/material';
import { useParams } from 'react-router-dom';
import { StyledDialog, FooterContainer } from '../../../styles';
import { UnmuteContext } from '../../NonEmptyMuted/MutedUser/MutedUser';
import Header from '../../../Header/Header';
import Footer from '../../../Footer/Footer';
import { MuteUnmuteUser } from '../MuteServer';

/**
 * Unmute Popup
 * @component
 * @property  {string} username the username of the muted user
 * @return {React.Component} - unmute popup component
 */

function UnmutePopUp(props) {
  const { subReddit } = useParams();
  const { userName } = props;
  const handleUnmute = () => {
    MuteUnmuteUser(userName, '', subReddit, 'unmute');
  };

  const {
    openUnmute, handleClickCloseUnmute,
  } = React.useContext(UnmuteContext);
  return (
    <StyledDialog
      fullScreen
      open={openUnmute}
      width="410px"
      height="190px"
    >
      <Box>
        <Header buttonFunction={handleClickCloseUnmute} headerText="Confirm" />
        <Divider />
      </Box>
      <Box sx={{ padding: '16px' }}>
        <Typography fontSize="14px" sx={{ margin: '10px 0px' }}>
          Are you sure you want to unmute
          {' '}
          {userName}
          ?
        </Typography>
      </Box>
      <FooterContainer>
        <Footer firstButtonFunction={handleClickCloseUnmute} firstButtonText="Cancel" secondButtonFunction={handleUnmute} secondButtonText="Unmute" />
      </FooterContainer>
    </StyledDialog>
  );
}

export default UnmutePopUp;
