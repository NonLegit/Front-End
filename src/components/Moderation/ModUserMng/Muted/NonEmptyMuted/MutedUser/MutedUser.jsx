/* eslint-disable react/jsx-no-constructed-context-values */
/* eslint-disable import/no-cycle */
import { Box, Typography } from '@mui/material';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import * as React from 'react';
import { useNavigate } from 'react-router-dom';
import {
  UserBar, UserContainer, UserMngButton, StyledAvatar, UsermngButtonContainer, TextContainer,
} from '../../../styles';
import MoreDetails from '../MoreDetails/MoreDetails';
import UnmutePopUp from '../../MuteUserPopUp/UnmutePopUp/UnmutePopUp';
import calculateTime from '../../../../../../utils/calculateTime';

export const UnmuteContext = React.createContext();

/**
 * Banned user instace
 * @component
 * @property  {object} user an object containing all information about the user from its username until the link to the its profile picture and the date that user is muted
 * @property {object} muteInfo contain the comment of the moderator on the muted user
 * @return {React.Component} - Banned user instace component
 */

function MutedUser(props) {
  const navigate = useNavigate();

  // navigate
  const handleClickUser = (userName) => {
    navigate(`/user/${userName}`);
  };
  const { user, muteInfo } = props;

  const [openUnmute, setOpenUnmute] = React.useState(false);

  const handleClickOpenUnmute = () => { setOpenUnmute(true); };
  const handleClickCloseUnmute = () => { setOpenUnmute(false); };

  const [isOpened, setOpen] = React.useState(false);
  const handleClick = () => {
    setOpen(!isOpened);
  };
  return (
    <>
      <UserBar>
        <UnmuteContext.Provider value={{
          openUnmute, handleClickCloseUnmute,
        }}
        >
          <UnmutePopUp userName={user.userName} />
        </UnmuteContext.Provider>
        <UserContainer onClick={() => handleClickUser(user.userName)}>
          <StyledAvatar src={user.profilePicture} variant="square" sx={{ width: '32px', height: '32px' }} />
          <Box>
            <Typography
              padding="8px"
              fontSize="15px"
              fontWeight="bold"
            >
              {user.userName}
            </Typography>
          </Box>
        </UserContainer>
        <TextContainer>
          <Typography
            padding="8px"
            fontSize="12px"
            color="#878A8C"
          >
            {calculateTime(user.joinDate)}
          </Typography>
        </TextContainer>
        <UsermngButtonContainer>
          <UserMngButton
            disableRipple
            disableFocusRipple
            onClick={handleClickOpenUnmute}
          >
            Unmute
          </UserMngButton>
          <UserMngButton
            disableRipple
            disableFocusRipple
            onClick={handleClick}
            endIcon={(isOpened) ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          >
            More Details
          </UserMngButton>
        </UsermngButtonContainer>
      </UserBar>
      <MoreDetails modNote={muteInfo.muteMessage} isOpened={isOpened} />
    </>
  );
}
export default MutedUser;
