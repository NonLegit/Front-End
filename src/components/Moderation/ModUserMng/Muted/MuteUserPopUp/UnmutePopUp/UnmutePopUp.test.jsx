import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import UnmutePopUp from './UnmutePopUp';
import MutedUser from '../../NonEmptyMuted/MutedUser/MutedUser';

const user = { userName: 'fady' };
const muteInfo = { muteMessage: 'Spammer' };

test('test snapshot', async () => {
  const tree = renderer.create(<Router><MutedUser user={user} muteInfo={muteInfo}><UnmutePopUp userName={user.userName} /></MutedUser></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
