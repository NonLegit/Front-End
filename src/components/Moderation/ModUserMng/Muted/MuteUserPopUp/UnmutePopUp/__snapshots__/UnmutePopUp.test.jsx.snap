// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  <div
    className="MuiBox-root css-13en8op"
  >
    <div
      className="MuiBox-root css-1gwqxiu"
      onClick={[Function]}
    >
      <div
        className="MuiAvatar-root MuiAvatar-square MuiAvatar-colorDefault css-w2kf69-MuiAvatar-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-0"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-gu63a4-MuiTypography-root"
        >
          fady
        </p>
      </div>
    </div>
    <div
      className="MuiBox-root css-1kazmbr"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-oa9s9h-MuiTypography-root"
      >
        a few seconds ago
      </p>
    </div>
    <div
      className="MuiBox-root css-pup5y"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-p5wwkm-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Unmute
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-p5wwkm-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        More Details
        <span
          className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="KeyboardArrowDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>,
  <div
    className="MuiBox-root css-21nkcj"
  >
    <div
      className="MuiBox-root css-1rzk8ps"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-16acbxi-MuiTypography-root"
      >
        mod note:
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1gaz5e1-MuiTypography-root"
      >
        Spammer
      </p>
    </div>
  </div>,
]
`;
